src/expander.ml

```diff
@@ -351,6 +358,36 @@ let expand_and_record_deps acc ~(dir : Path.Build.t) ~read_package ~dep_kind
       ~targets_written_by_user ~map_exe ~expand_var ~cc
       t pform syntax_version =
   let res =
+    let module Multiplicity = struct
+      type t = Multiple | One
+    end
+    in
+    let targets ~(multiplicity : Multiplicity.t) =
+      let loc = String_with_vars.Var.loc pform in
+      begin match (targets_written_by_user : Targets.t) with
+      | Infer ->
+        Errors.fail loc "You cannot use %s with inferred rules."
+          (String_with_vars.Var.describe pform)
+      | Forbidden context ->
+        Errors.fail loc "You cannot use %s in %s."
+          (String_with_vars.Var.describe pform) context
+      | Static { targets = l; multiplicity = declared_multiplicity } ->
+        let value = Value.L.dirs l (* XXX hack to signal no dep *) in
+        match multiplicity, declared_multiplicity, l with
+        | Multiple, _, _
```

Comment:
Shouldn't `Multiple, One, _` be an error as well?

Reply:
Made that change.